### PR TITLE
[Woo POS] M2: Add modal banner for simple products only

### LINKED ISSUE #13357: [Woo POS] M2: Simple products only modal
Continuation of https://github.com/woocommerce/woocommerce-ios/issues/13317 , the designs show a different banner once the first one has been dismissed. This modal will appear when the merchant taps `?` 

Screenshot 2024-07-17 at 12 53 26

### PR DESCRIPTION
**Pull request by soundsokay**
The PR adds a new modal view, `SimpleProductsModalView`, that provides details about only simple products being supported by Woo POS.

The new view may be triggered from the existing header banner that states "Showing simple products only" when POS mode is entered (see #13317).

Closing the header banner causes an info icon to be shown in the Products header. Tapping that will display the new modal banner for simple products.

The PR also centralizes modal presentation logic in PointOfSaleDashboardViewModel (commit 1f4084b3e9a1ea31096b73ec121555ca54e2013c).

The new view doesn't yet match the design in a pixel-perfect manner per prior agreement this week. The link on the modal view to "Create an order in store management" also doesn't yet work. These can be taken up under a separate ticket.

Closes: #13357

## Testing information

- Enter POS mode.
- Observe the "Showing simple products only" banner is shown.
- Tap the banner or the banner's info icon.
-  Observe the modal is show with "Why can't I see my products?" as the title.
- Click OK or the X icon to close the modal.
- Click the X icon in the header banner.
- Observe an info icon appears in the Products header.
- Tap the info icon.
- Observe the "Why can't I see my products?" modal is shown.

## Screenshots

https://github.com/user-attachments/assets/3edd4439-b633-46ee-aa50-c4e2e5703f95

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift [hunk 1]
```diff
@@ -5,30 +5,44 @@ struct ItemListView: View {
     @ScaledMetric private var scale: CGFloat = 1.0
     @ObservedObject var viewModel: ItemListViewModel
     @Environment(\.floatingControlAreaSize) var floatingControlAreaSize: CGSize
+    @State private var showModal: Bool = false
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/Presentation/ItemListView.swift]: Maybe `showSimpleProductsModal`? Just in case we add more/different modals.

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift [hunk 2]
```diff
@@ -47,23 +61,22 @@ private extension ItemListView {
             HStack {
                 headerTextView
                 Spacer()
-                Button(action: {
-                    // TODO:
-                    // https://github.com/woocommerce/woocommerce-ios/issues/13357
-                    debugPrint("Not implemented")
-                }, label: {
-                    Image(uiImage: .infoImage)
-                })
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/Presentation/ItemListView.swift]: Why is this removed? There should be an info button when that initial banner is removed.
In https://github.com/woocommerce/woocommerce-ios/pull/13417/files I started updating product list and left a method `openInfoBanner()` that could call `showModal.toggle()` and we can call it from this button and when tapping the whole banner card

Screenshot 2024-07-25 at 10 18 52

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift [hunk 3]
```diff
@@ -5,30 +5,44 @@ struct ItemListView: View {
...
+
+            if showModal {
+                SimpleProductsModalView(isPresented: $showModal)
+                    .background(Color.clear)
+                    .transition(.move(edge: .bottom))
+                    .onAppear {
+                        withAnimation(.easeInOut) {
+                            showModal = true
+                        }
+                    }
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/Presentation/ItemListView.swift]: I suppose we need to show this one and other modals above the main ZStack in the POS, because now it is showing on top of products and messes up the layout. We could use one of our view models to set which modal should be shown on top of entire POS UI

Simulator Screenshot - iPad Pro (11-inch) (4th generation) 17 2 - 2024-07-25 at 10 22 15

Screenshot 2024-07-25 at 10 22 12

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift (hunk 2 above)
Comment (soundsokay) [reply WooCommerce/Classes/POS/Presentation/ItemListView.swift]: I added it back. Looks like we both worked on the same code. I incorporated your styling code into mine during resolving merge conflicts.

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift (hunk 3 above)
Comment (soundsokay) [reply WooCommerce/Classes/POS/Presentation/ItemListView.swift]: This has been addressed.

Comment (soundsokay) [reply WooCommerce/Classes/POS/Presentation/ItemListView.swift]: > We could use one of our view models to set which modal should be shown on top of entire POS UI

Good idea. Implemented in commit 1f4084b3e9a1ea31096b73ec121555ca54e2013c.

File: WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift [hunk 4]
```diff
@@ -36,6 +36,22 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
...
+
+    @Published var activeModal: ModalType? = nil
+
+    enum ModalType: Identifiable {
+        case simpleProducts
+        // Add other modals if needed
+
+        var id: String {
+            switch self {
+            case .simpleProducts:
```
Comment (bozidarsevo) [inline WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift]: Good thing there is no default case. When we add new cases the compiler will show us where we need to add some logic :) p2: pfoUAQ-gM-p2

File: WooCommerce/Classes/POS/Presentation/ItemListView.swift (hunk 3 above)
Comment (soundsokay) [reply WooCommerce/Classes/POS/Presentation/ItemListView.swift]: I needed to back this out and re-merge trunk. An earlier merge of trunk led to a tangle that I thought might need generics to sort out.

Suggest we can take up centralizing modal view presentation in a later PR.

Comment (soundsokay) [reply WooCommerce/Classes/POS/Presentation/ItemListView.swift]: >We could use one of our view models to set which modal should be shown on top of entire POS UI

I added this back in commit a5df2a7be8514ddbc2d37972a7712304d2321f1d

File: WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift [hunk 5]
```diff
@@ -37,6 +37,17 @@ struct PointOfSaleDashboardView: View {
                     .shadow(color: Color.black.opacity(0.08), radius: 4)
                     .offset(x: Constants.floatingControlHorizontalOffset, y: -Constants.floatingControlVerticalOffset)
                     .trackSize(size: $floatingSize)
+                    .zIndex(1)
+            }
+
+            if viewModel.shouldShowModal(.simpleProducts) {
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift:43]: I read the discussion with @bozidarsevo but I couldn't understand 100% why we cannot use native sheet presentation logic? 

It will also streamline some of the design work (modal size, rounded corners, titles), rather than showing a custom view on top of a `ZStack`?

If I understood correctly, the concern may have been that we show pop up on top of product list, rather than on top of the whole pos but we could implement the logic in the dashboard view and do the same as with the card reader pop-up:
`.sheet(isPresented: $totalsViewModel.showsCardReaderSheet`?

Are there any blockers around this?

File: WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift [hunk 6]
```diff
@@ -37,6 +37,17 @@ struct PointOfSaleDashboardView: View {
...
+                SimpleProductsModalView(isPresented: Binding(
+                    get: { viewModel.shouldShowModal(.simpleProducts) },
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift:45]: May be enough to simplify it to `viewModel.showsSimpleProductsSheet` if we use the native approach?

File: WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift [hunk 7]
```diff
@@ -130,7 +136,7 @@ private extension PointOfSaleDashboardView {
     }
 
     var productListView: some View {
-        ItemListView(viewModel: itemListViewModel)
+        ItemListView(viewModel: self.itemListViewModel, dashboardViewModel: viewModel)
```
Comment (staskus) [inline WooCommerce/Classes/POS/Presentation/PointOfSaleDashboardView.swift:139]: Although we haven't yet found a good approach for our architecture (https://github.com/woocommerce/woocommerce-ios/issues/13441), maybe we could pass [a binding](https://developer.apple.com/documentation/swiftui/binding#overview) here rather than a whole dashboardViewModel? 

The idea being that `ItemListView` could change `isShowing = true` and then `PointOfSaleDashboardView` would react accordingly.

**Review by staskus — COMMENTED** [review]
Thanks for the work!

My main question comes from the decision to implement custom modals rather than relying on native sheets.  As I understand it comes from the previous discussion but I want to make sure I understand the reasoning.

Plus, modals are managed by ItemListViewModel while shown on PointOfSaleDashboardView so there may be an opportunity to sort out the dependencies.

**Comment by staskus** [comment]
Based on modal work in other places https://github.com/woocommerce/woocommerce-ios/pull/13466#discussion_r1698732237 I understand that native sheets do not satisfy all our requirements.

**Comment by joshheald** [comment]
@soundsokay I'm taking a look at this issue, as we have some new modal code, so I'm going to make it work with that... hope that's OK. I'll close this PR for clarity.
